engine/paymail_servant.go

```diff
@@ -0,0 +1,69 @@
+package engine
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/mrz1836/go-cachestore"
+)
+
+var (
+	ErrCapabilitiesPkiUnsupported  = errors.New("server doesn't support PKI")
+	ErrCapabilitiesPikeUnsupported = errors.New("server doesn't support PIKE")
+)
+
+type PaymailServant struct {
```

Comment:
As in some previous comment, I though that API of this struct will be rather available on go-paymail

Reply:
I thought about it and even start implementation this way. But I like possibility to have cache and do not want to force it in go-paymail